x-pack/plugins/security_solution/public/flyout/right/components/threat_intelligence_overview.tsx

```diff
@@ -0,0 +1,134 @@
+
+/**
+ * Threat Intelligence section under Insights section, overview tab.
+ * The component fetches the necessary data, then pass it down to the InsightsSubSection component for loading and error state,
+ * and the SummaryPanel component for data rendering.
+ */
+export const ThreatIntelligenceOverview: React.FC = () => {
+  const { eventId, indexName, dataFormattedForFieldBrowser } = useRightPanelContext();
+  const { openLeftPanel } = useExpandableFlyoutContext();
+  const { isAlert } = useBasicDataFromDetailsData(dataFormattedForFieldBrowser);
+
+  const goToThreatIntelligenceTab = useCallback(() => {
+    openLeftPanel({
+      id: LeftPanelKey,
+      path: LeftPanelInsightsTabPath,
+      params: {
+        id: eventId,
+        indexName,
+      },
+    });
+  }, [eventId, openLeftPanel, indexName]);
+
+  // retrieve the threat enrichment fields with value for the current document
+  // (see https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/common/cti/constants.ts#L35)
+  const eventFields = useMemo(
+    () => getEnrichmentFields(dataFormattedForFieldBrowser || []),
+    [dataFormattedForFieldBrowser]
+  );
+
+  // retrieve existing enrichment fields and their value
+  const existingEnrichments = useMemo(
+    () =>
+      isAlert
+        ? parseExistingEnrichments(dataFormattedForFieldBrowser || []).map((enrichmentData) =>
+            timelineDataToEnrichment(enrichmentData)
+          )
+        : [],
+    [dataFormattedForFieldBrowser, isAlert]
+  );
+
+  // api call to retrieve all documents that match the eventFields
+  const { result: enrichmentsResponse, loading: isEnrichmentsLoading } =
+    useInvestigationTimeEnrichment(eventFields);
+
+  // combine existing enrichment and enrichment from the api response
+  // also removes the investigation-time enrichments if the exact indicator already exists
+  const allEnrichments = useMemo(() => {
+    if (isEnrichmentsLoading || !enrichmentsResponse?.enrichments) {
+      return existingEnrichments;
+    }
+    return filterDuplicateEnrichments([...existingEnrichments, ...enrichmentsResponse.enrichments]);
+  }, [isEnrichmentsLoading, enrichmentsResponse, existingEnrichments]);
+
+  // separate threat matches (from indicator-match rule) from threat enrichments (realtime query)
+  const {
+    [ENRICHMENT_TYPES.IndicatorMatchRule]: threatMatches,
+    [ENRICHMENT_TYPES.InvestigationTime]: threatEnrichments,
+  } = groupBy(allEnrichments, 'matched.type');
+
+  const threatMatchesCount = (threatMatches || []).length;
+  const threatEnrichmentsCount = (threatEnrichments || []).length;
+
+  const data: InsightsSummaryPanelData[] = [
+    {
+      icon: 'image',
+      value: threatMatchesCount,
+      text: threatMatchesCount <= 1 ? THREAT_MATCH_DETECTED : THREAT_MATCHES_DETECTED,
+    },
+    {
+      icon: 'warning',
+      value: threatEnrichmentsCount,
+      text: threatMatchesCount <= 1 ? THREAT_ENRICHMENT : THREAT_ENRICHMENTS,
+    },
+  ];
```

Comment:
looks like it could go into some custom hook, adjacent to the component code :) have you considered that?

Reply:
I wasn't convinced at first, but after extracting the code and writing the unit tests I like it